agent/utils/retry/retry.go

```diff
@@ -15,13 +15,28 @@ package retry
 
 import (
 	"context"
+	"sync"
+	"time"
 
 	apierrors "github.com/aws/amazon-ecs-agent/agent/api/errors"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
 	"github.com/aws/amazon-ecs-agent/agent/utils/ttime"
 )
 
 var _time ttime.Time = &ttime.DefaultTime{}
 
+type TaskEventsFlowController struct {
+	flowControl      map[string]chan bool
```

Comment:
Maybe I'm misunderstanding, but why is this a channel? I don't fully see where we are using channel concepts with this object. Since we already have the mutex, couldn't this just be `map[string]bool` ?